src/controller.js

```diff
@@ -0,0 +1,47 @@
+(function(exports){
+
+  var newsLog
+  var summaryrequest
+  var newsview
+
+  function newsLog(){
+    return newslog
+  }
+
+  function summaryRequest(){
+    return summaryrequest
+  }
+
+  function newsView(){
+    return newsview
+  }
+
+  function showSelectedSummary(){
+    window.addEventListener("hashchange", findSummaryUrl)
+  }
+
+  function findSummaryUrl(){
+    requestSummary(findIndex(window.location))
+  }
+
+  function findIndex(location){
+    return location.hash.split("#")[1];
+  }
+
+  function requestSummary(index){
+    var articleUrl = newslog.articles[index].url
+    summaryrequest = new ArticleSummaryRequest(articleUrl)
+    summaryrequest.requestSummary(newsView().displayArticleSummary)
+  }
+
+  function onLoad(){
+    newslog = new NewsLog()
+    newsview = new NewsHeadlinesView(newsLog())
+    newslog.getArticles(newsView().displayHeadLines)
+    showSelectedSummary()
+  }
+
+  exports.onLoad = onLoad
```

Comment:
It feels like what you really want to export here is a Controller class - a lot of the logic could be a in Controller but some of it belongs elsewhere. 

A good pattern is to have an app.js file which handles application setup. Then rather than <body onload="onload()"></body>, I would remove the onload event from the HTML and in your app.js have some code that looks a bit like:

```
function onLoad(){
  newslog = new NewsLog()
  newsview = new NewsHeadlinesView(newsLog)
  controller = new Controller(newsview);
  controller.initialise();
}

if (document.readyState === 'complete') {
  onload();
} else {
  document.onload = onload;
}
```
Your controller can then contain other initialisation code relating to the different views (e.g. showSelectedSummary and newslog.getArticles etc). You're effectively distinguishing here between application *setup* (create all your objects and inject relevant dependencies) and then initialising the application *flow* (getting data, rending html etc).